include/drivers/spi.h

```diff
@@ -129,11 +129,16 @@ extern "C" {
  *    to act as a CS line
  * @param delay is a delay in microseconds to wait before starting the
  *    transmission and before releasing the CS line
+ * @param gpio_dt_flags is the devicetree flags corresponding to how the CS
+ *    line should be driven. Currently ACTIVE_LOW / ACTIVE_HIGH is ignored
+ *    in favor of the SPI_CS_ACTIVE_HIGH and SPI_CS_ACTIVE_LOW options
+ *    on struct spi_config.
  */
 struct spi_cs_control {
 	struct device	*gpio_dev;
 	uint32_t		gpio_pin;
 	uint32_t		delay;
+	uint8_t			gpio_dt_flags;
```

Comment:
That's bloating the struct by 4bytes due to alignment.

Looking at gpio.h it now seems like pin is always of type gpio_pin_t which is... a byte. And you are not using the right type here also. So rework this to: 
```
struct spi_cs_control {
        struct device     *gpio_dev;
        uint32_t          delay;
        gpio_pin_t        gpio_pin;
        gpio_dt_flags_t   gpio_dt_flags;
        uint16_t          _unused;
};
```

Reply:
Thanks for the feedback, the original plan was to keep the types consistent in the locations that changes were made. I will make the changes.